### ISSUE TITLE
Downscaled sprite rendering pixelated 

### ISSUE THREAD

**Issue by alaa-eddine** [pixijs/pixi.js#1802]
Hello, 
is there a way to get good rendering for downscaled sprite ?
I use unique size textures and adapt to device size by scaling my sprites.
but when I go down the rendering becomes very bad .

bitmapfonts are the worst, almost unreadable. normal images becomes very pixelated whatever the configuration I use  ; I tried many configurations for antialias and forceFXAA ...

is it an issue or normal behaviour ?

do someone have any suggestion to enhance this ?

to get an idea : 
this is how the normal size looks like : http://labs.ezelia.com/pixi/fullsize.png
and here is the downscaled version : http://labs.ezelia.com/pixi/tiny.png

**Comment by JiDW** [comment]
You should use different resolutions and downscale (or upscale) from the nearest resolution you made available to achieve fullscreen. TexturePacker is a software able to export to a format supported by PIXI and it can automatically export your assets to different resolution. (It is called "Scalling Variant" inside the software)

For text, you should use the resolution property of a PIXI.Text object or use dynamic font size by multiplying pixel font size (10px instead of 20px)

For bitmap fonts, you need to export multiple size (ex : 10px and 20px) and select the best for your screensize. In our game, we have 35 bitmapfonts (mostly numbers, so small textures) to deal with our needs. You can put bitmapfonts inside a spritesheet (with texturepacker) thanks to PIXI v3.

Hope this help. 

TL;DR : multiple resolutions, select the best then fill the screen with a small scale.

**Comment by GoodBoyDigital** [comment]
Hi @alaa-eddine ,

One way is to use mipmapping, this will let the gpu do the work and create a bunch of smaller assets for you. The only catch is that the texture must be a power of two size.

```
var myPO2Texture = PIXI.Texture.fromImage(''powerOfTwo.png");
myPO2Texture.baseTexture.mipmap = true;
```
hope this helps! 

Thanks

**Comment by grovesNL** [comment]
@GoodBoyDigital Mipmap only seems to be available to WebGLRenderer. Should CanvasRenderer support this as well?

Alternatively, is there an efficient way to perform texture swapping for mipmaps (ie. switch from 64x64 to 128x128 on a zoom event, when many sprites share the same texture)? I could use multiple textures and call `setTexture` on each sprite for each zoom event, but this approach seems wasteful.
